app/models/data_set.rb

```diff
@@ -65,8 +65,12 @@ def places_for_postcode(postcode, distance = nil, limit = nil)
       end
     else
       location_data = MapitApi.location_for_postcode(postcode)
-      location = Point.new(latitude: location_data.lat, longitude: location_data.lon)
-      places_near(location, distance, limit)
+      if location_data.lat && location_data.lon
```

Comment:
OR... add a method to validate `long` and `lat` to the `Point` class and have something like:
`location.valid_lat_long? ? places_near(location, distance, limit) : []`
I'm probably overthinking this way too much... :)